kubernetes-client/src/main/java/io/fabric8/kubernetes/client/osgi/ManagedKubernetesClient.java

```diff
@@ -155,9 +158,9 @@
     @Reference(referenceInterface = ExtensionAdapter.class, cardinality = ReferenceCardinality.OPTIONAL_MULTIPLE, policy = ReferencePolicy.DYNAMIC, bind = "bindExtensionAdapter", unbind = "unbindExtensionAdapter"),
     @Reference(referenceInterface = OAuthTokenProvider.class, cardinality = ReferenceCardinality.OPTIONAL_UNARY, policyOption = ReferencePolicyOption.GREEDY, bind = "bindOAuthTokenProvider", unbind = "unbindOAuthTokenProvider")
 })
-public class ManagedKubernetesClient extends BaseClient implements NamespacedKubernetesClient {
+public class ManagedKubernetesClient implements NamespacedKubernetesClient {
 
-  private NamespacedKubernetesClient delegate;
+  private DefaultKubernetesClient delegate;
```

Comment:
Maybe it might make sense to expose the adapters or registration capabilities through an interface and we don't have to depend on the specific type.

Reply:
Since Client methods are currently re-exposed on every Client subclass, I was wanting to keep the methods there to a minimum.  I'll update the managed openshift one though to just use OpenShiftClient.